OrbitBase/CMakeLists.txt

```diff
@@ -0,0 +1,31 @@
+cmake_minimum_required(VERSION 3.15)
+
+project(OrbitBase)
+
+add_library(OrbitBase INTERFACE)
+
+target_compile_features(OrbitBase INTERFACE cxx_std_17)
```

Comment:
Is there a reason why we do this only for certain component and not for all of Orbit?

Reply:
I don't know. I think that we have just been doing this when necessary while leaving the other modules untouched.

Reply:
We do this because of semantical reasons.
`target_compile_features(OrbitBase INTERFACE cxx_std_17)` says that this module needs AT LEAST C++17.

Independently from this, you usually lock your project's C++ version by setting `CMAKE_CXX_STANDARD` in your main CMakeLists.txt file. Setting this variable to `14` (C++14) would immediately give you a configuration error.

Having these kind of semantic helpers might not be very important for a small project like Orbit but it is good
practice. It makes a lot more sense for bigger projects, with hundreds of targets.